app/models/widget.rb

```diff
@@ -0,0 +1,16 @@
+class Widget < ActiveRecord::Base
+  enum widget_type: Constants::WIDGET_TYPES
```

Comment:
Yes, but how will an engine, which is a pluggable Rails application, inject its own widget Strings into `Constants::WIDGET_TYPES`? If you freeze this variable, any calls (e.g. `Constants::WIDGET_TYPES << :my_new_widget`) will fail.